lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
> This PR is removing the usage of Route.prototype.serialize if the serializer is already present (from the Router.map which is always available synchronously).

Maybe I'm reading it wrong, but let me try to explain the problem I have in mind. When the serializer isn't defined in Router.map, the route instance (handler) defines `serialize` and we lazy-load it. Before the route (handler) is loaded, we will fallback to the default serializer behavior, when the route is loaded, we will use `handler.serialize` instead.